src/main/scala/Plugin.scala

```diff
@@ -1,460 +0,0 @@
-            } else {
-              println(s"Missing bintray credentials $credsFile. Some bintray features depend on this.")
-              None
-            }
-          case creds => creds
-        })
-
-  def bintrayPublishSettings: Seq[Setting[_]] = bintrayCommonSettings ++ Seq(
-    credentialsFile in bintray in Global := Path.userHome / ".bintray" / ".credentials",
-    name in bintray := moduleName.value,
-    bintrayOrganization in bintray in Global := { if (sbtPlugin.value) Some("sbt") else None },
-    // todo: don't force this to be sbt-plugin-releases
-    repository in bintray in Global := { if (sbtPlugin.value) "sbt-plugin-releases" else "maven" },
-    publishMavenStyle := {
-      if (sbtPlugin.value) false else publishMavenStyle.value
-    },
-    vcsUrl in bintray := resolveVcsUrl.recover {
-      case NonFatal(e) =>
-        None
-    }.get,
-    packageLabels in bintray := Nil,
-    description in bintray <<= description,
-    // note: publishTo may not have dependencies. therefore, we can not rely well on inline overrides
-    // for inline credentials resolution we recommend defining bintrayCredentials _before_ mixing in the defaults
-    // perhaps we should try overriding something in the publishConfig setting -- https://github.com/sbt/sbt-pgp/blob/master/pgp-plugin/src/main/scala/com/typesafe/sbt/pgp/PgpSettings.scala#L124-L131
-    publishTo in bintray <<= publishToBintray,
-    resolvers in bintray <<= appendBintrayResolver,
-    credentials in bintray <<= (credentialsFile in bintray).map {
-      Credentials(_) :: Nil
-    },
-    packageAttributes in bintray <<= (sbtPlugin, sbtVersion) {
-      (plugin, sbtVersion) =>
-        if (plugin) Map(AttrNames.sbtPlugin -> Seq(Attr.Boolean(plugin)))
-        else Map.empty
-    },
-    versionAttributes in bintray <<= (crossScalaVersions, sbtPlugin, sbtVersion) {
-      (scalaVersions, plugin, sbtVersion) =>
-        val sv = Map(AttrNames.scalas -> scalaVersions.map(Attr.Version(_)))
-        if (plugin) sv ++ Map(AttrNames.sbtVersion-> Seq(Attr.Version(sbtVersion))) else sv
-    },
-    omitLicense in bintray in Global := { if (sbtPlugin.value) sbtPlugin.value else false },
-    ensureLicenses <<= ensureLicensesTask,
-    ensureCredentials <<= ensureCredentialsTask,
-    ensureBintrayPackageExists <<= ensurePackageTask,
-    publishVersionAttributes <<= publishVersionAttributesTask,
-    unpublish in bintray <<= unpublishTask.dependsOn(ensureBintrayPackageExists, ensureLicenses),
-    remoteSign in bintray <<= remoteSignTask,
-    syncMavenCentral in bintray <<= syncMavenCentralTask
-  ) ++ Seq(
-    resolvers <++= resolvers in bintray,
-    credentials <++= credentials in bintray,
-    publishTo <<= publishTo in bintray,
-    // We attach this to publish configruation, so that publish-signed in pgp plugin can work.
-    publishConfiguration <<= publishConfiguration.dependsOn(ensureBintrayPackageExists, ensureLicenses), 
```

Comment:
I'll have another look though all of the changes tomorrow. I'd like to publish this tonight, but I keep getting held up by gotchas ( its reminding me of sonatype my sonatype days! ) The original idea behind this plugin was to remove as many steps between compiling code to publishing the results as possible giving the community a "share" button for scala libraries. I'd like not to have the first experience be an error message, followed by a series of settings overrides. In my idea world the defaults should *just work*. I understand the want for more control but more control should mean more configuration. Less control should be no configuration. Not the other way around. I'm really happy with the overall set of changes but I kind of wish we did this is smaller iterations :/ 

In any case I'll see if I can at least get the ensurePackageTask to work with publish tomorrow night.